cmd/agent/commands/grpc.go

```diff
@@ -75,20 +83,32 @@ func BuildGrpcCmd(env runtime.Environment, config *agent.Config) *cobra.Command
 			return agent.ApplyDisruption(cmd.Context(), disruptor, duration)
 		},
 	}
+
 	cmd.Flags().DurationVarP(&duration, "duration", "d", 0, "duration of the disruptions")
 	cmd.Flags().DurationVarP(&disruption.AverageDelay, "average-delay", "a", 0, "average request delay")
 	cmd.Flags().DurationVarP(&disruption.DelayVariation, "delay-variation", "v", 0, "variation in request delay")
 	cmd.Flags().Int32VarP(&disruption.StatusCode, "status", "s", 0, "status code")
 	cmd.Flags().Float32VarP(&disruption.ErrorRate, "rate", "r", 0, "error rate")
 	cmd.Flags().StringVarP(&disruption.StatusMessage, "message", "m", "", "error message for injected faults")
-	cmd.Flags().StringVarP(&iface, "interface", "i", "eth0", "interface to disrupt")
-	cmd.Flags().UintVarP(&port, "port", "p", 8080, "port the proxy will listen to")
-	cmd.Flags().UintVarP(&target, "target", "t", 80, "port the proxy will redirect request to")
 	cmd.Flags().StringSliceVarP(&disruption.Excluded, "exclude", "x", []string{}, "comma-separated list of grpc services"+
 		" to be excluded from disruption")
-	cmd.Flags().BoolVar(&transparent, "transparent", true, "run as transparent proxy")
-	cmd.Flags().StringVar(&upstreamHost, "upstream-host", "localhost",
-		"upstream host to redirect traffic to")
+	cmd.Flags().BoolVar(&transparent, "transparent", true,
+		"Run as transparent proxy. This mode requires root or NET_ADMIN privileges")
+	// 120, 107, and 54 are ASCII for xk6, respectively.
+	cmd.Flags().StringVar(&transparentAddress, "transparent-address", "127.120.107.54",
+		"When running in transparent mode, this address will be added to the interface specified in "+
+			"--transparent-interface. "+
+			"Proxy will use this address to send requests upstream. "+
+			"This flag is ignored if the agent does not run in transparent mode.")
+	cmd.Flags().StringVar(&transparentInterface, "transparent-interface", "lo",
+		"When running in transparent mode, the agent will add the address specified in --transparent-address to "+
+			"this interface. "+
+			"This flag does not affect in which interface the proxy listens in, which is only determined by --listen. "+
+			"This flag is ignored if the agent runs in non-transparent mode.")
+	cmd.Flags().StringVarP(&proxyConfig.ListenAddress, "listen", "l", ":8080",
+		"Address where the proxy will listen at")
+	cmd.Flags().StringVarP(&proxyConfig.UpstreamAddress, "target", "t", "localhost:80",
+		"Address where the proxy will redirect requests to.")
```

Comment:
Originally, I was hoping for this PR to remove the need to listen on all interfaces, thus the change from ports to addresses. Unfortunately, due to reasons explained in the comments in `iptables.go`, this turned out not to be possible.

I think addresses are more convenient to specify than ports and IPs separately, but this change is now out of the scope of the PR. I can revert it if needed.